SvgImage.js

```diff
@@ -1,46 +1,82 @@
+const fetchResultDebug = require("debug")("react-native-remote-svg:SvgImage:fetch:result");
+import Promise from 'bluebird';
+Promise.config({ cancellation: true }); // Need to explicitly enable this feature
 
 const firstHtml =
   '<html><head><style>html, body { margin:0; padding:0; overflow:hidden; background-color: transparent; } svg { position:fixed; top:0; left:0; height:100%; width:100% }</style></head><body>';
 const lastHtml = '</body></html>';
 
 class SvgImage extends Component {
-  state = { fetchingUrl: null, svgContent: null };
+
+  static propTypes = {
+    source: PropTypes.shape({
+        uri: PropTypes.string.isRequired
+      }).isRequired,
+    containerStyle: PropTypes.oneOfType([
+        PropTypes.instanceOf(StyleSheet).isRequired,
+        PropTypes.object.isRequired,
+      ]),
+    style: PropTypes.oneOfType([
+        PropTypes.instanceOf(StyleSheet).isRequired,
+        PropTypes.object.isRequired,
+    ]),
+  }
+
+  state = {
+    svgContent: null,
+    fetchingPromise: null,
+  };
+
   componentDidMount() {
-    this.doFetch(this.props);
+    this.doFetch();
+  }
+
+  componentWillUnmount() {
+    const { fetchingPromise } = this.state || {};
+    if(fetchingPromise) fetchingPromise.cancel();
   }
-  componentWillReceiveProps(nextProps) {
-    this.doFetch(nextProps);
+
+  componentDidUpdate(prevProps) {
+    const { source:oldSource } = prevProps || {};
+    const { uri:oldUri } = oldSource || {};
+    const { source:newSource } = this.props || {};
+    const { uri:newUri } = newSource || {};
+    if(oldUri !== newUri) this.doFetch();
   }
-  doFetch = props => {
-    let uri = props.source && props.source.uri;
+
+  doFetch() {
+    const { source } = this.props || {};
+    const { uri } = source || {};
     if (uri) {
       if (uri.match(/^data:image\/svg/)) {
         const index = uri.indexOf('<svg');
-        this.setState({ fetchingUrl: uri, svgContent: uri.slice(index) });
+        this.setState({ svgContent: uri.slice(index) });
       } else {
-        console.log('fetching', uri);
-        fetch(uri)
-          .then(res => res.text())
-          .then(text => {
-            this.setState({ fetchingUrl: uri, svgContent: text });
-          })
-          .catch(err => {
-            console.error('got error', err);
-          });
+        fetchingDebug('Fetching SVG from %s', uri);
+        this.setState(({fetchingPromise:previousFetch}) => ({ fetchingPromise:
+          Promise.resolve(fetch(uri))
+            .call("text")
+            .then(text => this.setState({ svgContent: text }))
+            .catch(e => console.error(`Error fetching SVG URI: ${e.message||e}`, {uri, e}))
+            .return((previousFetch && previousFetch.isPending()) ? previousFetch : null) // Ensure we resolve/cancel previous fetch
+        }))
       }
     }
-  };
+  }
+
   render() {
-    const props = this.props;
-    const { svgContent } = this.state;
-    if (svgContent) {
+    const props = this.props || {};
```

Comment:
this.props should always be an object in react. Have you seen any situation where it might be falsy? It shouldn't be